driver/src/price_finding/naive_solver.rs

```diff
@@ -0,0 +1,100 @@
+use crate::models;
+
+use super::price_finder_interface::Solution;
+use web3::types::U256;
+use crate::models::Order;
+use itertools::Itertools;
+use std::error::Error;
+
+enum OrderPairType {
+    TypeIa,
+    TypeIb,
+    TypeII
+}
+
+impl Order {
+    fn matches(&self, other: &Order) -> bool {
+        self.opposite_tokens(other) && self.have_price_overlap(other)
+    }
+
+    fn match_compare(&self, other: &Order) -> Option<OrderPairType> {
+        if self.matches(other) {
+            if self.buy_amount <= other.sell_amount && self.sell_amount <= other.buy_amount {
+                Some(OrderPairType::TypeIa)
+            } else if self.buy_amount >= other.sell_amount && self.sell_amount >= other.buy_amount {
+                Some(OrderPairType::TypeIb)
+            } else {
+                Some(OrderPairType::TypeII)
+            }
+        }
+        None()
+    }
+
+    fn opposite_tokens(&self, other: &Order) -> bool {
+        self.buy_token == other.sell_token && self.sell_token == other.buy_token
+    }
+
+    fn have_price_overlap(&self, other: &Order) -> bool {
+        self.buy_amount * other.buy_amount <= other.sell_amount * self.sell_amount
+    }
+}
+
+pub fn solve(orders: &Vec<models::Order>, num_tokens: u8) -> Result<Solution, Error> {
```

Comment:
I think it would be good to solve function, which is also processing the account balances. If I am not mistaken, then we also want to have a valid solution, which fits to our accounting system.

Reply:
Yes, I suppose that would be true. However, I think I will first focus on getting a complete solution first. I haven't coded the surplus in Rust just yet. I would also like to somehow lose the x, y loop that I have here.

Reply:
Yes, just to clarify, as I may have misunderstood your suggestion here is to pass account balances into this function so that we know what we are about to execute was actually a valid order (i.e. the user has sufficient balance to fulfill the executed sell amount).